src/date-range-input/date-range-input.tsx

```diff
@@ -380,15 +438,18 @@ export const DateRangeInput = ({
     };
 
     const handleCalendarSelect = (val: string) => {
+        console.log("coming here");
         if (currentFocus === "start") {
             handleStartDateChange(val);
         } else if (currentFocus === "end") {
             handleEndDateChange(val);
         }
-
         if (isWeekSelection) {
             handleWeekSelectionChange(val);
         }
+        if (isFixedRangeSelection) {
+            handleFixedRangeSelectionChange(val);
+        }
```

Comment:
this could be converted to a switch case for variant i.e.

```
switch (variant)
  case fixed-range
    handleFixedRangeSelectionChange
  case week
    handleWeekSelectionChange
  default
    ...
```